pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -243,6 +243,26 @@ public static class Broker {
         Math.max(1, Math.min(10, Runtime.getRuntime().availableProcessors() / 2));
     // Same logic as CombineOperatorUtils
 
+    // Config for Jersey ThreadPoolExecutorProvider.
+    // By default, Jersey uses the default unbounded thread pool to process queries.
+    // By enabling it, BrokerManagedAsyncExecutorProvider will be used to create a bounded thread pool.
+    public static final String CONFIG_OF_ENABLE_BOUNDED_HTTP_ASYNC_EXECUTOR =
+            "pinot.broker.enable.bounded.http.async.executor";
+    public static final boolean DEFAULT_ENABLE_BOUNDED_HTTP_ASYNC_EXECUTOR = false;
+    // Default capacities for the bounded thread pool
+    public static final String CONFIG_OF_HTTP_ASYNC_EXECUTOR_MAX_POOL_SIZE =
+            "pinot.broker.http.async.executor.max.pool.size";
+    public static final int DEFAULT_HTTP_ASYNC_EXECUTOR_MAX_POOL_SIZE =
+            Runtime.getRuntime().availableProcessors() * 2;
+    public static final String CONFIG_OF_HTTP_ASYNC_EXECUTOR_CORE_POOL_SIZE =
+            "pinot.broker.http.async.executor.core.pool.size";
+    public static final int DEFAULT_HTTP_ASYNC_EXECUTOR_CORE_POOL_SIZE =
+            Runtime.getRuntime().availableProcessors() * 2;
+    public static final String CONFIG_OF_HTTP_ASYNC_EXECUTOR_QUEUE_SIZE =
+            "pinot.broker.http.async.executor.queue.size";
+    public static final int DEFAULT_HTTP_ASYNC_EXECUTOR_QUEUE_SIZE =
```

Comment:
queue size actually is bounded by memory. but I think availableProcessors() * 2 is OK.

Reply:
Would the default queue size of availableProcessors() * 2 being too small? During spiky request we may drop requests prematurely?

Reply:
Hi Jasper, I didn't test the default values so they might be small. Can you share what might be a better number here?

Reply:
Yes that looks good to me. Thanks!